### PR TITLE
[12.0][FIX] website_snippet_carousel_product: Hide warning message

### PR DESCRIPTION
**Pull request by Tardo**
Hide warning message to public users.

cc @tecnativa TT26155

@chienandalu @CarlosRoca13

### CODE REVIEW

File: website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js [hunk 1]
```diff
@@ -51,20 +52,22 @@ odoo.define("website_snippet_carousel_product.s_product_carousel", function (req
...
+                                self.$target.html(
+                                    $("<div/>", {class: "col-md-6 offset-md-3"}).append(
+                                        $("<div/>", {
+                                            class:
+                                                "alert alert-warning" +
+                                                " alert-dismissible text-center",
+                                            text: _t(
+                                                "No products was found." +
+                                                " Make sure you have products" +
+                                                " published on the website."
```
Comment (pedrobaeza) [inline website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: Maybe better?

```suggestion
                                                " published on the website that matches the domain."
```

File: website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js [hunk 2]
```diff
@@ -51,20 +52,22 @@ odoo.define("website_snippet_carousel_product.s_product_carousel", function (req
...
-                                            " alert-dismissible text-center",
-                                        text: _t(
-                                            "No products was found." +
-                                            " Make sure you have products" +
-                                            " published on the website."
-                                        ),
-                                    })
-                                )
-                            );
+                            if (!session.is_website_user) {
```
Comment (pedrobaeza) [inline website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: This excludes also portal users?

Comment (Tardo) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: No, portal users will see the warning message.

Comment (pedrobaeza) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: They shouldn't. Only users with website group must see it (those that are able to modify the domain).

**Comment by Tardo** [comment]
@pedrobaeza Changes done

File: website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js (hunk 2 above)
Comment (pedrobaeza) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: Is this comment attended?

Comment (Tardo) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: no

Comment (pedrobaeza) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: Well, it's important, as most e-commerce users are portal ones, so they will see it anyway.

Comment (Tardo) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: Done, now only users with "website.group_website_designer" group will see warning messages

**Comment by Tardo** [comment]
Changes done

**Comment by Tardo** [comment]
Travis green

**Review by CarlosRoca13 — APPROVED** [review]
:+1: LGTM

File: website_snippet_carousel_product/templates/snippet.xml [hunk 3]
```diff
@@ -38,49 +38,54 @@
...
-                    data-slide="next"
-                    t-if='num_slides > 1'
-                >
-                    <i class="fa fa-2x fa-chevron-right text-black-50" />
-                    <span class="sr-only">Next</span>
-                </a>
-            </div>
+            </t>
+            <t t-if="is_website_designer and not len(objects)">
+                <div class="alert alert-warning alert-dismissible text-center">No products was found. Make sure you have products published on the website that matches the domain.</div>
```
Comment (pedrobaeza) [inline website_snippet_carousel_product/templates/snippet.xml:87]: This text seems not translatable.

**Comment by Tardo** [comment]
Translations updated

File: website_snippet_carousel_product/templates/snippet.xml (hunk 3 above)
Comment (pedrobaeza) [reply website_snippet_carousel_product/templates/snippet.xml:87]: I'm afraid there's a typo:
```suggestion
                <div class="alert alert-warning alert-dismissible text-center">No products were found. Make sure you have products published on the website that match the domain.</div>
```

**Comment by chienandalu** [comment]
Was this addressed in later versions?

**Comment by pedrobaeza** [comment]
@Tardo can you answer?

**Pull request closed without merging**